op-node/sources/caching/cache.go

```diff
@@ -22,6 +22,18 @@ func (c *LRUCache) Get(key any) (value any, ok bool) {
 	return value, ok
 }
 
+func (c *LRUCache) GetOrPeek(key any, usePeek bool, recordMetrics bool) (value any, ok bool) {
+	if usePeek {
+		value, ok = c.inner.Peek(key)
+	} else {
+		value, ok = c.inner.Get(key)
+	}
+	if c.m != nil && recordMetrics {
+		c.m.CacheGet(c.label, ok)
+	}
+	return value, ok
+}
+
```

Comment:
LGTM
and one more question, before, we made a specific [PreFetchCache](https://github.com/bnb-chain/opbnb/blob/36d34ab9fd7b3cd48399c1431763a9cc69243802/op-service/sources/eth_client.go#L181) for receipts caching, so i wonder why not reuse LRUCache with this `GetOrPeek` method as well, but to create a new one  [(PreFetchCache)](https://github.com/bnb-chain/opbnb/blob/36d34ab9fd7b3cd48399c1431763a9cc69243802/op-service/sources/eth_client.go#L181) ?

Reply:
[PreFetchCache](https://github.com/bnb-chain/opbnb/blob/36d34ab9fd7b3cd48399c1431763a9cc69243802/op-service/sources/eth_client.go#L181) is an first-in-first-out queue that allows us to actively remove data from it. It is more suitable for our use case here: when the safe block height is processed, the L1 data it depends on is no longer needed and can be deleted. At the same time, when the cache is full, preFetch should stop working and wait for the cache to be consumed.
Originally, all caches should use PreFetchCache, but only receipts data can be used as the key with blockNumber, while other data cannot. The inability to use blockNumber as a key means that PreFetchCache cannot be used. However, it doesn't matter, as long as receipts are used, it can control the rate of the preFetch goroutine.